src/geom/face_tri6.C

```diff
@@ -174,6 +174,17 @@ dof_id_type Tri6::key (const unsigned int s) const
 
 
 
+unsigned int Tri6::which_node_am_i(unsigned int side,
+                                   unsigned int side_node) const
+{
+  libmesh_assert_less (side, this->n_sides());
+  libmesh_assert_less (side_node, 3);
+
+  return Tri6::side_nodes_map[side][side_node];
+}
+
```

Comment:
Well, one thing is that not all the asserts are the same (they depend on the Elem type), so there's not a ton of duplicate code savings there. Second, I thought about adding a virtual function that just returned the respective `Foo::side_nodes_map[s][n]` but decided that was about the same number of lines of code as this solution...